### PR TITLE
scalability: intro

### LINKED ISSUE #11: Write down the first version of Scalability
Reference

* https://tikv.org/deep-dive/scalability/introduction/
* https://tikv.org/deep-dive/scalability/horizontal-or-vertical/

### PR DESCRIPTION
**Pull request by pingyu**
### What issue does this PR solve?

- close #11 

### What is changed:
Write down the first version of Scalability.
https://github.com/pingyu/tikv-dev-guide/blob/scalability-intro/src/understanding-tikv/scalability/intro.md

### CODE REVIEW

**Comment by tisonkun** [comment]
Thanks for your contribution. I'm going through now.

**Review by tisonkun — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
Sorry there is nothing to comment >_<

File: src/understanding-tikv/scalability/intro.md [hunk 1]
```diff
@@ -1 +1,21 @@
 # Scalability
+
+In the database field, `Scalability` is the term we use to describe the capability of a system to handle a growing amount of work. Even if a system is working reliably and fast today, it doesn't mean it will necessarily work well in the future, especially when the increased load exceeds what the system can process. In modern systems, the amount of data we handle can far outgrow our original expectations, so scalability is a critical consideration for the design of a database.
+
+There are two main types of scalability:
+
+1. Vertical scaling, which is also known as __scaling up__, means adding resources to single node in a system, typically involving the improvement of CPU, memory or storage, to become a more powerful single computer. Vertical scaling is limited by the technology of semiconductor industry, and the cost per hertz of CPU or byte of memory/storage will increase dramatically when near the technical limit.
```
Comment (andylokandy) [inline src/understanding-tikv/scalability/intro.md:7]:
```suggestion
1. Vertical scaling, which is also known as __scaling up__, means adding resources to a single node in a system, typically involving the improvement of CPU, memory, or storage, to become a more powerful single computer. Vertical scaling is limited by the technology of the semiconductor industry, and the cost per hertz of CPU or byte of memory/storage will increase dramatically when near the technical limit.
```

File: src/understanding-tikv/scalability/intro.md [hunk 2]
```diff
@@ -1 +1,21 @@
...
+
+2. Horizontal scaling, which is also known as __scaling out__, means adding more machines to a system and distributing the load across multiple smaller machines. As computer prices have dropped and performance continues to increase, high-performance computing applications have adopted low-cost commodity systems for tasks. System architects may deploy hundreds of small computers in a cluster, to obtain aggregate computing power that far more powerful than system based on stand-alone single computer. Moreover, with widespread use of [Cloud computing](https://en.wikipedia.org/wiki/Cloud_computing) technology, horizontal scalable is necessary for a system to adapt to the resiliency of Cloud.
```
Comment (andylokandy) [inline src/understanding-tikv/scalability/intro.md:9]:
```suggestion
2. Horizontal scaling, which is also known as __scaling out__, means adding more machines to a system and distributing the load across multiple smaller machines. As computer prices have dropped and performance continues to increase, high-performance computing applications have adopted low-cost commodity systems for tasks. System architects may deploy hundreds of small computers in a cluster, to obtain aggregate computing power that is far more powerful than a system based on a stand-alone single computer. Moreover, with the widespread use of [Cloud computing](https://en.wikipedia.org/wiki/Cloud_computing) technology, horizontal scalable is necessary for a system to adapt to the resiliency of Cloud.
```

**Comment by pingyu** [comment]
Thanks for your review. Suggestions applied, PTAL~ @andylokandy

**Review by andylokandy — APPROVED** [review]

**Comment by andylokandy** [comment]
@pingyu Thank you!

**Pull request merged**